src/resolvers/registries/npm-resolver.js

```diff
@@ -36,17 +42,21 @@ export default class NpmResolver extends RegistryResolver {
     const satisfied = await config.resolveConstraints(Object.keys(body.versions), range);
     if (satisfied) {
       return body.versions[satisfied];
-    } else {
-      const versions = Object.keys(body.versions);
-      throw new MessageError(
-        config.reporter.lang(
-          'couldntFindVersionThatMatchesRange',
-          body.name,
-          range,
-          (versions.length > 20) ? versions.join(os.EOL) : versions.join(', '),
-        ),
-      );
+    } else if (request && request.resolver && request.resolver.activity) {
+      request.resolver.activity.end();
+      config.reporter.log(config.reporter.lang('couldntFindVersionThatMatchesRange', body.name, range));
+      const response = await inquirer.prompt([{
+        name: 'package',
+        type: 'list',
+        message: config.reporter.lang('chooseVersionFromList'),
+        choices: Object.keys(body.versions).reverse(),
+        pageSize: 20,
```

Comment:
I think a dynamic page size would be better here. Something like https://github.com/yarnpkg/yarn/blob/master/src/cli/commands/upgrade-interactive.js#L27-L30